Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -345,31 +346,43 @@ class ReferenceValue : public FieldValue::BaseValue {
     if (type() != other.type()) return false;
 
     auto& other_value = Cast<ReferenceValue>(other);
-    return database_id_ == other_value.database_id_ && key_ == other_value.key_;
+    return database_id() == other_value.database_id() &&
+           key() == other_value.key();
   }
 
   ComparisonResult CompareTo(const BaseValue& other) const override {
     ComparisonResult cmp = CompareTypes(other);
     if (!util::Same(cmp)) return cmp;
 
     auto& other_value = Cast<ReferenceValue>(other);
-    cmp = Compare(database_id_, other_value.database_id_);
+    cmp = Compare(database_id(), other_value.database_id());
     if (!util::Same(cmp)) return cmp;
 
-    return Compare(key_, other_value.key_);
+    return Compare(key(), other_value.key());
   }
 
   std::string ToString() const override {
-    return absl::StrCat("Reference(key=", key_.ToString(), ")");
+    return absl::StrCat("Reference(key=", key().ToString(), ")");
```

Comment:
Question: it is deliberate not to include the database ID here, right?

Reply:
Correct. This preserves the existing behavior.